plugin/src/main/java/org/opensearch/ml/action/tasks/CancelBatchJobTransportAction.java

```diff
@@ -177,25 +184,39 @@ private void processRemoteBatchPrediction(MLTask mlTask, ActionListener<MLCancel
         RemoteInferenceInputDataSet inferenceInputDataSet = new RemoteInferenceInputDataSet(parameters, ActionType.BATCH_PREDICT_STATUS);
         MLInput mlInput = MLInput.builder().algorithm(FunctionName.REMOTE).inputDataset(inferenceInputDataSet).build();
         String modelId = mlTask.getModelId();
+        User user = RestActionUtils.getUserContext(client);
 
         try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
             ActionListener<MLModel> getModelListener = ActionListener.wrap(model -> {
-                if (model.getConnector() != null) {
-                    Connector connector = model.getConnector();
-                    executeConnector(connector, mlInput, actionListener);
-                } else if (clusterService.state().metadata().hasIndex(ML_CONNECTOR_INDEX)) {
-                    ActionListener<Connector> listener = ActionListener
-                        .wrap(connector -> { executeConnector(connector, mlInput, actionListener); }, e -> {
-                            log.error("Failed to get connector " + model.getConnectorId(), e);
-                            actionListener.onFailure(e);
-                        });
-                    try (ThreadContext.StoredContext threadContext = client.threadPool().getThreadContext().stashContext()) {
-                        connectorAccessControlHelper
-                            .getConnector(client, model.getConnectorId(), ActionListener.runBefore(listener, threadContext::restore));
+                modelAccessControlHelper.validateModelGroupAccess(user, model.getModelGroupId(), client, ActionListener.wrap(access -> {
+                    if (!access) {
+                        actionListener.onFailure(new MLValidationException("You don't have permission to cancel this batch job"));
+                    } else {
+                        if (model.getConnector() != null) {
+                            Connector connector = model.getConnector();
+                            executeConnector(connector, mlInput, actionListener);
+                        } else if (clusterService.state().metadata().hasIndex(ML_CONNECTOR_INDEX)) {
+                            ActionListener<Connector> listener = ActionListener
+                                .wrap(connector -> { executeConnector(connector, mlInput, actionListener); }, e -> {
+                                    log.error("Failed to get connector " + model.getConnectorId(), e);
+                                    actionListener.onFailure(e);
+                                });
+                            try (ThreadContext.StoredContext threadContext = client.threadPool().getThreadContext().stashContext()) {
+                                connectorAccessControlHelper
```

Comment:
Do we need to check connector access control again here? Are model/connector all access controlled through the model group so ideally any role would have the same permission to access both models and connectors?

Reply:
I think its fine. batch_predict also only checks for only model access for now. Since predict API is controlled by model access, it makes sense to have it for batch job APIs too. If we are planning to have connector access too, then we should also include it into predict APIs for consistency.

And yes, I don't think connector access is controlled by model group access control. They were designed separately. I think we have to either unify them or remove one of them for remote models. Does not make sense to have both at connector and model level.